code/iaas/allocator/src/main/java/io/cattle/platform/allocator/service/impl/AllocatorServiceImpl.java

```diff
@@ -704,6 +704,21 @@ private String getHostUuid(Instance instance) {
     @SuppressWarnings("unchecked")
     private void callExternalSchedulerToReserve(AllocationAttempt attempt, AllocationCandidate candidate) {
         List<Long> agentIds = getAgentResource(attempt.getAccountId(), attempt.getInstances());
+        if (agentIds.isEmpty() && !attempt.getInstances().get(0).getSystem() && allocatorDao.schedulerServiceEnabled(attempt.getAccountId())
+                && allocatorDao.isScheulderIpsEnabled(attempt.getAccountId())) {
+            List<ResourceRequest> resourceRequests = gatherResourceRequests(attempt.getInstances(), attempt.getVolumes(), null);
+            for (ResourceRequest r : resourceRequests) {
+                if (r instanceof PortBindingResourceRequest) {
+                    PortBindingResourceRequest pr = (PortBindingResourceRequest)r;
+                    for (PortSpec ps : pr.getPortRequests()) {
+                        if (StringUtils.isEmpty(ps.getIpAddress()) || !"0.0.0.0".equals(ps.getIpAddress())) {
```

Comment:
We can probably use !StringUtils.equals("0.0.0.0", ps.getIpAddress()). It should cover both cases